README.md

```diff
@@ -1,14 +1,175 @@
+		return param * 2;
+	});
+</script>
+```
+
+OR
+
+```ts
+<script>
+	import { task } from 'svelte-concurrency';
+
+	const my_task = task(async function* (param: number) {
+		await new Promise((r) => setTimeout(r, 2000));
+		return param * 2;
+	}, { kind: 'default' });
+</script>
+```
+
+### Restartable task
+
+This will cancel the oldest instance of the task and start a new instance of it. You can also provide a `max` that will only restart the oldest task instance if the threshold is exceeded.
+
+```ts
+<script>
+	import { task } from 'svelte-concurrency';
+
+	const my_task = task.restart(async function* (param: number) {
+		await new Promise((r) => setTimeout(r, 2000));
+		return param * 2;
+	}, { max: 3 });
+</script>
+```
+
+OR
+
+```ts
+<script>
+	import { task } from 'svelte-concurrency';
+
+	const my_task = task(async function* (param: number) {
+		await new Promise((r) => setTimeout(r, 2000));
+		return param * 2;
+	}, { kind: 'restart', max: 3 });
+</script>
+```
+
+Both of the above will result in 3 simultaneous tasks being allowed to run. Triggering the task a fourth time will cancel the oldest task.
+
+### Droppable task
+
+This will cancel any new instances of the task. You can also provide a `max` that will only drop the task instances if the threshold is exceeded.
+
+```ts
+<script>
+	import { task } from 'svelte-concurrency';
+
+	const my_task = task.drop(async function* (param: number) {
+		await new Promise((r) => setTimeout(r, 2000));
+		return param * 2;
+	}, { max: 3 });
+</script>
+```
+
+OR
+
+```ts
+<script>
+	import { task } from 'svelte-concurrency';
+
+	const my_task = task(async function* (param: number) {
+		await new Promise((r) => setTimeout(r, 2000));
+		return param * 2;
+	}, { kind: 'drop', max: 3 });
+</script>
+```
+
+Both of the above will result in 3 simultaneous tasks being allowed to run. Triggering the task a fourth time will be cancelled and won't retry automatically.
+
+### Enqueue task
+
+This will add all task instances to a list and each task will be run in order. You can also provide a `max` that will dictate the number of task instances that will run at the same time.
+
+```ts
+<script>
+	import { task } from 'svelte-concurrency';
+
+	const my_task = task.enqueue(async function* (param: number) {
+		await new Promise((r) => setTimeout(r, 2000));
+		return param * 2;
+	}, { max: 3 });
+</script>
+```
+
+OR
+
+```ts
+<script>
+	import { task } from 'svelte-concurrency';
+
+	const my_task = task(async function* (param: number) {
+		await new Promise((r) => setTimeout(r, 2000));
+		return param * 2;
+	}, { kind: 'enqueue', max: 3 });
+</script>
+```
+
+Both of the above will result in 3 simultaneous tasks being allowed to run. Any additional instances of the task will be added to a list and run when there is space in the queue.
+
+### Accessing the task in the template
+
+As the return value from the task wrapper is a store, you can access it just like you would with any other store:
```

Comment:
We should probably mention that if you return something from the task you can await the `task.perform` to get the result (but also that if you don't await it the return value of `task.perform` is the same store.